### ISSUE TITLE
Why mapEventToState have currentState?

### ISSUE THREAD

**Issue by GiancarloCode** [felangel/bloc#161]
Hello

Why do you use `currentState` in `mapEventToState` if `currentState` is a property of `bloc` and can be access with the same way without be a parameter of `mapEventToState`

Thanks.

```dart
 @override
  Stream<int> mapEventToState(int currentState, CounterEvent event) async* {
    yield currentState +1;
  }
```
the same as this
```dart
 @override
  Stream<int> mapEventToState( CounterEvent event) async* {
    yield currentState +1;
  }
```

**Comment by felangel** [comment]
@GiancarloCode thanks for bringing this up! Originally, blocs didn't have a currentState property so the `mapEventToState` signature needed it. Now that `currentState` is a property in all Blocs it is no longer needed. #162 will address this change.

Thanks again for bringing this up! I totally forgot to make this simplification after adding `currentState`.

**Comment by felangel** [comment]
published [bloc v0.11.0](https://pub.dartlang.org/packages/bloc) 🎉
